samples/aws-samples/mnist-kmeans-sagemaker/kmeans-hpo-pipeline.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+
+import kfp
+from kfp import components
+from kfp import dsl
+from kfp.aws import use_aws_secret
+
+sagemaker_hpo_op = components.load_component_from_file('../../../components/aws/sagemaker/hyperparameter_tuning/component.yaml')
+
+@dsl.pipeline(
+    name='MNIST HPO test pipeline',
+    description='SageMaker hyperparameter tuning job test'
+)
+def hpo_test(region='us-west-2',
+    hpo_job_name='HPO-kmeans-sample',
+    image='',
+    algorithm_name='K-Means',
+    training_input_mode='File',
+    metric_definitions='{}',
+    strategy='Bayesian',
+    metric_name='test:msd',
+    metric_type='Minimize',
+    early_stopping_type='Off',
+    static_parameters='{"k": "10", "feature_dim": "784"}',
+    integer_parameters='[{"Name": "mini_batch_size", "MinValue": "450", "MaxValue": "550"}, \
+                         {"Name": "extra_center_factor", "MinValue": "10", "MaxValue": "20"}]',
+    continuous_parameters='[]',
+    categorical_parameters='[{"Name": "init_method", "Values": ["random", "kmeans++"]}]',
+    channels='[{"ChannelName": "train", \
```

Comment:
Minor: Have you considered to transform channels to simpler configs? If I am a user, I would think this is easy to break. If S3Uri, S3DataType amd S3DataDistributionType are fields helpful, can we think of a way to build abstraction?  This is low priority